.github/workflows/build-image.yaml

```diff
@@ -9,43 +9,72 @@ on:
     merge_group:
 
 jobs:
+    changes:
+        runs-on: ubuntu-latest
+        outputs:
+            should_skip: ${{ steps.check.outputs.should_skip }}
+        steps:
+            - uses: actions/checkout@v4
+            - uses: dorny/paths-filter@v3
+              id: filter
+              with:
+                  filters: |
+                      docs_only:
+                          - 'docs-v2/**'
+            - name: Determine if should skip
+              id: check
+              run: |
+                  # Skip if:
+                  # - only docs were changed
+                  # - not on master branch (always building image on master so we can deploy any commit)
+                  if [[ "${{ steps.filter.outputs.docs_only }}" == "true" && "${{ github.ref }}" != "refs/heads/master" ]]; then
+                      echo "should_skip=true" >> $GITHUB_OUTPUT
+                  else
+                      echo "should_skip=false" >> $GITHUB_OUTPUT
+                  fi
```

Comment:
leftover or it's missing on every other yaml?

Reply:
it is only in this one because the condition is more complex (ie: still build image on master even when docs-only ) so I encapsulated it a step but the other workflow will be skipped (no tests on master if docs-only changes for instance)